addons/account/models/partner.py

```diff
@@ -821,23 +821,38 @@ def _unlink_if_partner_in_account_move(self):
         if moves:
             raise UserError(_("The partner cannot be deleted because it is used in Accounting"))
 
-    def _increase_rank(self, field, n=1):
-        if self.ids and field in ['customer_rank', 'supplier_rank']:
+    def _increase_rank(self, field: str, n: int = 1):
```

Comment:
Here is how I would separate the functions

```python
    def _get_rank_cache(self, field):
        key = f'account.res.partner.increase_rank.{field}'
        postcommit = self.env.cr.postcommit
        if key in postcommit.data:
            return postcommit.data[key]
        cache = postcommit.data[key] = defaultdict(int)
    	@postcommit.add(...)
    	return cache
    
    def _increase_rank(self, field: str, n: int = 1):
        cache = self._get_rank_cache(field)
        for record in self.sudo():
            # comment...
            if record[field]:
                cache[record.id] += n
            else:
                record[field] += n
```

Reply:
I would avoid manipulate the cache if possible and I find it cleaner to leave everything in a single function.